include/wlr/types/wlr_output.h

```diff
@@ -303,26 +294,90 @@ void wlr_output_lock_attach_render(struct wlr_output *output, bool lock);
  * a lock.
  */
 void wlr_output_lock_software_cursors(struct wlr_output *output, bool lock);
+
 /**
- * Renders software cursors. This is a utility function that can be called when
- * compositors render.
+ * Hardware cursors:
+ *
+ * Hardware cursors are a feature of GPUs that allow a small framebuffer
+ * to be displayed over the main framebuffer. This is more efficient than
+ * drawing onto the main framebuffer using e.g. OpenGL.
+ *
+ * Not every backend/output will support hardware cursors, so you must always
+ * be prepared to fall back to rendering the cursor to the main framebuffer
+ * yourself.
+ *
+ * Hardware cursors are always in output buffer coordinates, and does not take
+ * scale nor rotation into account. You are responsible for doing this
+ * yourself.
+ *
+ * Hardware cursors are always in the ARGB8888 format.
  */
-void wlr_output_render_software_cursors(struct wlr_output *output,
-	pixman_region32_t *damage);
 
+/**
+ * Sets the size of the hardware cursor buffer to the values pointed to by
+ * x and y.
+ *
+ * Some backends have strict requirements for the cursor size and may not be
+ * able to use the size you requested exactly. The actual value used will be
+ * left in x and y. If this is not large enough for the cursor you wish to
+ * draw, you should fall back to using software cursors.
+ *
+ * This returns false if the backend does not support hardware cursors or from
+ * some other internal error. Either way, you should fall back to using
+ * software cursors.
+ */
+bool wlr_output_cursor_try_set_size(struct wlr_output *output, int *x, int *y);
```

Comment:
Let's also remove `_try` from the name, we don't have any precedent of doing this for things which might fail.